src/cls/queue/cls_queue_types.h

```diff
@@ -20,18 +20,23 @@ struct cls_queue_entry
 {
   ceph::buffer::list data;
   std::string marker;
+  bool needs_migration = false;
```

Comment:
why do we need to change the cls_queue entry type?
this is an abstract queue, unrelated to notifications

Reply:
please break to a separate vector of booleans
this is just used in-memoery, no need to encode/decode

Reply:
sure